File: solutions/Javascript/Find Peak Element.js
var findPeakElement = function (arr) {
  if (arr.length === 1) {
    return 0;
  }
  let left = 0;
  let right = arr.length - 1;
  while (left <= right) {
    let mid = Math.floor((left + right) / 2);
    if (mid === 0 && arr[mid] > arr[mid + 1]) {
      return mid;
    }
    if (mid === arr.length - 1 && arr[mid] > arr[mid - 1]) {
      return mid;
    }
    if (arr[mid] > arr[mid - 1] && arr[mid] > arr[mid + 1]) {
      return mid;
    }
    if (arr[mid] < arr[mid + 1]) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }
};

File: solutions/cpp/Find Peak Element.cpp
#include <vector>
class Solution
{
public:
    int findPeakElement(std::vector<int> &nums)
    {
        if (nums.size() == 1)
            return 0;
        int left = 0, right = nums.size() - 1, n = nums.size() - 1;
        while (left <= right)
        {
            int mid = (left + right) / 2;
            if (mid == 0 && nums[mid] > nums[mid + 1])
                return mid;
            if (mid == n && nums[mid] > nums[mid - 1])
                return mid;
            if ((mid > 0 && mid < n) && (nums[mid] > nums[mid - 1] && nums[mid] > nums[mid + 1]))
                return mid;
            if (nums[mid] < nums[mid + 1])
                left = mid + 1;
            else
                right = mid - 1;
        };
        return -1;
    }
};